### PR TITLE
Feature/rails capstone project

### PR DESCRIPTION
**Pull request by oguntayomicheal**
### IN this PR, the following milestones were completed;
- Splash screen
-  Sign up and log in pages
- Home page (categories page)
-  Transactions page
-  "Add a new category" page
-  "Add a new transaction" page
- Add  unit and integration tests

### CODE REVIEW

File: app/views/splash/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<%= render partial: 'shared/nav', locals: { h1_content: "Budget-App" } %>
+<div class='splash'>
+    <h1>Welcome to Budget-App</h1>
+    <p>Get started by signing up or logging in:</p>
+    <div id='splash_link'>
+        <a href="<%= new_user_registration_path %>">Sign Up</a>
+        <a href="<%= new_user_session_path %>">Log In</a>
+    </div>
```
Comment (elisha2kyakpo1) [inline app/views/splash/index.html.erb]: - [ ] Kindly style the sign-in/sign-up links to match the given design, I have attached an image for your reference.


splash


signup

File: app/views/products/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+</button>
+
+<div class='pro_cat'>
+  <h2>
+    <%= @category_name%>  
+  </h2>
+  <p class='cat-amount'>
+   - Total: $ <%= @products.sum(:amount)%>
+  </p>
+</div>
```
Comment (elisha2kyakpo1) [inline app/views/products/index.html.erb]: - [ ] Kindly position this div at the top and align the category name and the sum (total) properly matching the given design.

your-total


total-tr

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+  end
+
+  authenticated :user do
+    root 'categories#index', as: :authenticated_root
+  end
+  devise_for :users
+  resources :users
+  resources :categories do
+    resources :products
+  end
```
Comment (elisha2kyakpo1) [inline config/routes.rb]: - [ ] Kindly only expose the routes you are using, in this case, `create, index, new` to be used.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @oguntayomicheal, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Tests passing👏
👏 The app works as expected.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 3 above)
Comment (oguntayomicheal) [reply config/routes.rb]: Hi @elisha2kyakpo1, thanks for the review

**Review by fobadara — APPROVED** [review]
Hi @oguntayomicheal

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Clean code
✅ Proper implementation of reviewer's suggestions

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**